go-controller/pkg/ovn/master.go

```diff
@@ -794,6 +821,15 @@ func (oc *Controller) ensureNodeLogicalNetwork(node *kapi.Node, hostSubnets []*n
 		}
 	}
 
+	if oc.loadBalancerGroupSupport {
```

Comment:
do you really need this block? You create the load balancer group in startClusterMaster, which is called before Run:
```
				if err := oc.StartClusterMaster(nodeName); err != nil {
					panic(err.Error())
				}
				if err := oc.Run(wg, nodeName); err != nil {
					panic(err.Error())
				}
```

Run is where we start watching nodes, so its not possible for you to add nodes before the cluster load balancer group exists.

Reply:
If the NB ovsdb server supports load balancer groups then we should attach the cluster-wide load balancer group to the node logical switch.  Otherwise, when running against an older OVN version, we're just going to use the legacy mode and attach load balancers directly to each node switch as there won't be any cluster-wide load balancer group that we can use.